api/v1/resourceset_types.go

```diff
@@ -65,6 +65,8 @@ type ResourceSetSpec struct {
 	Wait bool `json:"wait,omitempty"`
 }
 
+// +kubebuilder:validation:XValidation:rule="self.name != '' || has(self.selector)", message="at least one of name or selector must be set for input provider references"
+// +kubebuilder:validation:XValidation:rule="self.name == '' || !has(self.selector)", message="cannot set both name and selector for input provider references"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="!has(self.name) || !has(self.selector)", message="at least one of name or selector must be set for input provider references"
// +kubebuilder:validation:XValidation:rule="has(self.name) || has(self.selector)", message="cannot set both name and selector for input provider references"
```

Reply:
Good catch! I have no idea why `go fmt ./...` transforms `''` in `”`.

Btw the logic in XValidation is inverted, for "at least one" we need "has(...) || has(...)" and for "cannot set both" we need "!has(...) || !has(...)"